json/pickler/src/test/scala/sttp/tapir/json/pickler/PicklerTest.scala

```diff
@@ -523,6 +502,21 @@ class PicklerTest extends AnyFlatSpec with Matchers {
     codec.decode(encoded) shouldBe Value(inputObj)
   }
 
+  it should "handle sealed hierarchies consisting of objects only" in {
```

Comment:
maybe we should somehow separate tests per-feature, here: for enumerations (direct, wrapped, customised)